security/token-file-provider.1.rst

```diff
@@ -0,0 +1,175 @@
+on pre-created files and is designed for maximum portability.
+token-file-provider takes a configuration file that specifies the services
+for which tokens shall be generated and the Vault access policy
+that shall be applied to those tokens.
+token-file-provider assumes that there is some underlying protection mechanism
+that will be used to prevent EdgeX services from reading each other’s tokens.
+
+
+OPTIONS
+=======
+  \-h, \--help
+    Display help text
+
+  \-c, \--confdir <confdir>
+    Look in this directory for configuration.toml instead.
+
+  \-p, \--profile <name>
+    Indicate configuration profile other than default
+
+
+FILES
+=====
+
+configuration.toml
+------------------
+This file specifies the TCP/IP location of the Vault service
+and parameters used for Vault token generation.
+
+::
+
+  [SecretService]
+  Server = "localhost"
+  Port = 8200 
+
+  [TokenFileProvider]
+  PrivilegedTokenPath = /run/edgex/secrets/token-file-provider/vault-token.json
+  ConfigFile = token-config.json
+  OutputDir = /run/edgex/secrets/
+  OutputFilename = vault-token.json
+
+
+vault-token.json
+----------------
+This file contains a token use to authenticate to Vault.
+
+::
+
+  {
+    "auth": {
+      "client_token": "s.wOrq9dO9kzOcuvB06CMviJhZ"
+    }
+  }
+
+
+token-config.json
+-----------------
+This configuration file tells token-file-provider which tokens to generate.
+
+In order to avoid a directory full of `.hcl` files,
+this configuration file uses the JSON serialization of HCL,
+documented at https://github.com/hashicorp/hcl/blob/master/README.md.
+
+::
+
+  {
+    "service-name": {
+      "edgex-use-defaults": true,
+      "custom_policy": [
+        {
+          "path": {
+            "secret/edgex/pki/tls/edgex-kong": {
+              "capabilities": [ "list", "read" ]
+            }
+          }
+        }
+      ],
+      "custom_token_parameters": { }
+    }
+  }
+
+
+When edgex-use-default is true (the default),
+the following is added to the policy specification
+for the auto-generated policy.
+Thus, the final policy created for the token will be the union
+of the policy below (if using the default policy)
+plus the ``custom_policy`` defined above.
+
+::
+
+  {
+    "path": {
+      "secret/edgex/service-name/*": {
+        "capabilities": [ "create", "update", "delete", "list", "read" ]
+      }
+    }
+  }
+
+When edgex-use-default is true (the default),
+the following is inserted (if not overridden) to the token parameters for the generated token.
+(See https://www.vaultproject.io/api/auth/token/index.html#create-token.)
+
+::
+
+  "display_name": service-name
+  "no_parent":    true
+  "policies":     [ "edgex-service-service-name" ]
+
+Note that ``display_name`` may be used by ``go-mod-secrets``
+as a hint for locating service secrets.
+
+
+{OutputDir}/{service-name}/{OutputFilename}
+-------------------------------------------
+For example: ``/run/edgex/secrets/edgex-kong/vault-token.json``
```

Comment:
do we want this to be "kong" or "security-proxy"?

Reply:
I think it should be `security-proxy` or even just `proxy-token.json` to keep it generically named

Reply:
Actually, I think the real name of the only secret consuming service today is `edgex-security-proxy-setup` so it would be `/run/edgex/secrets/edgex-security-proxy-setup/vault-token.json`.

Reply:
Fixed.